utility/cancel_requests_x_days_old

```diff
@@ -0,0 +1,219 @@
+             var waitText = blob[data.recordID].blockingStepID == 0 ? 'Pending ' : 'Waiting for ';
+             var status = '';
+             if(blob[data.recordID].stepID == null && blob[data.recordID].submitted == '0' && blob[data.recordID].lastStatus == null) {
+                 status = '<span style="color: #e00000">Not Submitted</span>';
+             }
+             else if(blob[data.recordID].stepID == null) {
+                 var lastStatus = blob[data.recordID].lastStatus;
+                 if(lastStatus == '') {
+                     lastStatus = '<a href="index.php?a=printview&recordID='+ data.recordID +'">Check Status</a>';
+                 }
+                 status = '<span style="font-weight: bold; color: #e00000;">' + lastStatus + '</span>';
+             }
+             else {
+                 status = waitText + blob[data.recordID].stepTitle;
+             }
+
+             if(blob[data.recordID].deleted > 0) {
+                 status += ', Cancelled';
+             }
+
+             $('#'+data.cellContainerID).html(status);
+             if(blob[data.recordID].userID == '<!--{$userID|unescape|escape:'quotes'}-->') {
+                 $('#'+data.cellContainerID).css('background-color', '#feffd1');
+             }
+         }}
+         ]);
+
+        var tGridData = [];
+        for(var i in res) {
+            tGridData.push(res[i]);
+        }
+        grid.setData(tGridData);
+        //grid.sort('recordID', 'desc');
+        grid.renderBody();
+        grid.announceResults();
+
+        $('#header_date').css('width', '60px');
+        $('#header_service').css('width', '150px');
+        $('#header_currentStatus').css('width', '100px');
+
+    });
+    
+    let leafSearch = new LeafFormSearch('searchContainer');
+    leafSearch.setSearchFunc(search => {
+        try {
+            leafSearchTerms = $.parseJSON(search);
+        }
+        catch(err) {
+            return;
+        }
+        
+        generateQuery(leafSearchTerms);
+
+        query.execute();
+    });
+    leafSearch.init();
+    document.querySelector('#'+ leafSearch.getPrefixID() + 'advancedSearchButton').click();
+}
+    
+function generateQuery(advSearch) {
+    query.clearTerms();
+    console.log(advSearch);
+    /*for(var i in advSearch) {
+        if(advSearch[i].id != 'data'
+           && advSearch[i].id != 'dependencyID') {
+            query.addTerm(advSearch[i].id, advSearch[i].operator, advSearch[i].match, advSearch[i].gate);
+        }
+        else {
+            query.addDataTerm(advSearch[i].id, advSearch[i].indicatorID, advSearch[i].operator, advSearch[i].match, advSearch[i].gate);
+        }
+    }*/
+    query.addDataTerm('stepID', '', '!=', 'submitted', 'AND');
+    query.addDataTerm('dateInitiated', '', '<=', getDaysAgo(numDays), 'AND');
+    query.addDataTerm('deleted', '', '=', 0, 'AND');
+}
+    
+function getDaysAgo(days){
```

Comment:
This function has a lot of vars in it.  Pretty sure it doesn't need them as it returns the value being asked for.

Reply:
Changed